File: domain/src/main/java/no/nav/melosys/domain/eessi/BucInformasjon.java
package no.nav.melosys.domain.eessi;

import java.time.LocalDate;
import java.util.List;
import java.util.Set;

public record BucInformasjon(String id, boolean erÅpen, String bucType,
                             LocalDate opprettetDato, Set<String> mottakerinstitusjoner,
                             List<SedInformasjon> seder) {

    public String getId() {
        return id;
    }

    public String getBucType() {
        return bucType;
    }

    public LocalDate getOpprettetDato() {
        return opprettetDato;
    }

    public Set<String> getMottakerinstitusjoner() {
        return mottakerinstitusjoner;
    }

    public List<SedInformasjon> getSeder() {
        return seder;
    }
}

File: domain/src/main/kotlin/no/nav/melosys/domain/eessi/BucInformasjon.kt
package no.nav.melosys.domain.eessi

import java.time.LocalDate

/**
 * Metadata om en BUC (Business Use Case) i EESSI-systemet.
 */
data class BucInformasjon(
    val id: String?,
    val erÅpen: Boolean,
    val bucType: String?,
    val opprettetDato: LocalDate,
    val mottakerinstitusjoner: Set<String>?,
    val seder: List<SedInformasjon>
) {
    /**
     * Eksplisitt metode for Java interop og bakoverkompatibilitet.
     * Java-kode forventer erÅpen() metode (ikke getErÅpen()).
     */
    fun erÅpen(): Boolean = erÅpen
}
